blockchain/genesis.go

```diff
@@ -136,6 +137,29 @@ func (e *GenesisMismatchError) Error() string {
 	return fmt.Sprintf("database already contains an incompatible genesis block (have %x, new %x)", e.Stored[:8], e.New[:8])
 }
 
+// findBlockWithState returns the latest block with state.
+func findBlockWithState(db database.DBManager) *types.Block {
+	headBlock := db.ReadBlockByHash(db.ReadHeadBlockHash())
+	if headBlock == nil {
+		logger.Crit("failed to read head block by head block hash")
+	}
+
+	startBlock := headBlock
+	for _, err := state.New(headBlock.Root(), state.NewDatabase(db)); err != nil; {
+		if headBlock.NumberU64() == 0 {
+			logger.Crit("failed to find state")
```

Comment:
How about leaving the log with `headBlock.Root()`?

Reply:
Good. I've updated.